pkg/blueprint/firstboot_customizations.go

```diff
@@ -0,0 +1,154 @@
+package blueprint
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+)
+
+type FirstbootCustomization struct {
+	Scripts []FirstbootScriptCustomization `json:"scripts,omitempty" toml:"scripts,omitempty"`
+}
+
+type FirstbootScriptCustomization struct {
+	union json.RawMessage
+}
+
+// FirstbootCommonCustomization contains common fields for all firstboot customizations.
+type FirstbootCommonCustomization struct {
+	// Type of the firstboot customization. Supported values are:
+	// "custom", "satellite", and "aap".
+	Type string `json:"type,omitempty" toml:"type,omitempty"`
+
+	// Optional firstboot name. Must be unique within the blueprint and only
+	// alphanumeric characters with dashes and underscores are allowed.
+	Name string `json:"name,omitempty" toml:"name,omitempty"`
+
+	// Ignore errors when executing the firstboot script and continue with
+	// execution of the following firstboot scripts, if any. By default,
+	// firstboot scripts are executed in order and if one of them fails, the
+	// execution stops immediately.
+	IgnoreFailure bool `json:"ignore_failure,omitempty" toml:"ignore_failure,omitempty"`
+}
+
+// CustomFirstbootCustomization contains fields specific to custom firstboot
+// customizations. All fields must be prefixed with "Custom".
```

Comment:
What does `All fields must be prefixed with "Custom".` mean?  Same for the other docstrings.  Is it left over from a previous version?